### ISSUE TITLE
[REQUEST FEATURE] No shutdown when SMART long test is running

### ISSUE THREAD

**Issue by revers59** [OpenMediaVault-Plugin-Developers/openmediavault-autoshutdown#52]
how to prevent autoshutdown from shutting down the server when a SMART test is running?

Smart long test can take 10 hours on a 4TB disk.

I think when smart test running no process can be detected.

We can use smartctl -a <device> for check the status

Self-test execution status:      ( 249) Self-test routine in progress...
                                        90% of test remaining.

Currently All my long smart test are reset by host because autoshutdown dont see activity.

**Comment by Ansem93** [comment]
I would love this!

**Comment by coelner** [comment]
Something like this [untested] @ryecoaaron 
```
################################################################
#
#   name         : _check_smarthdd
#   parameter      : none
#   global return   : none
#   return         : 1      : if any hdd is running smart test no shutdown
#               : 0      : if no hdd is runngin smart test
#
# This script checks if any drive runs a smart test
# if yes -> no shutdown, next cycle
# if no -> next check and shutdown if all cycles failed
#
_check_smart()
{
	# ## disabled for testing
	# _log "DEBUG: _check_smart() disabled for testing"
	# return 0

	RVALUE=0
	# loop through /dev
	for dev in /dev/sd[a-z]; do 
		smartctl -a -n standby $dev | grep -q "in_progress"; 
		if [ $? -ne 0 ]; then
			_log "INFO: $dev no SMART test active."
		else
			let RVALUE++
			_log "INFO: $dev drive runs a SMART test -  No shutdown"
		fi
	done;

	if $DEBUG ; then _log "DEBUG: _check_smart(): RVALUE: $RVALUE" ; fi

	return ${RVALUE}
}
```

**Comment by nrandon** [comment]
This feature is now available in the current release 5.1.7 closing off
